### ISSUE TITLE
Extract StdinDiscarder into separate library

### ISSUE THREAD

**Issue by ffMathy** [sindresorhus/ora#143]
It would be super useful to use separately.

**Comment by sindresorhus** [comment]
// @stroncium

**Comment by stroncium** [comment]
Yeah, no prob. Though probably on next weekend, have to make money for now :cry:.

**Comment by Richienb** [comment]
@sindresorhus To speed up the completion of this issue, I'll extract the code if you wish.

**Comment by sindresorhus** [comment]
I would prefer if @stroncium did it as he's the one with the most experience about all the edge-cases.

**Comment by stroncium** [comment]
Yeah, that sounds like a good idea, that thing can be quite tricky. Sorry for delay, a lot of irl stuff happening. I'll get to it asap.

**Comment by sindresorhus** [comment]
@stroncium I have now extracted it to a separate package: https://github.com/sindresorhus/stdin-discarder

However, I noticed some things:
- The following seems to be dead code. There's nothing that uses it: https://github.com/sindresorhus/stdin-discarder/blob/a367682430700a0528905b7dda045f1dbf87a3b8/index.js#L17-L32
- After extracting the code, I noticed that it does not actually prevent stdin even in Ora before extracting the code. However, it seems the stdin prevention worked before https://github.com/sindresorhus/ora/pull/135.
